debian/postnews.1
.TH POSTNEWS 1 "2002-06-28"
.SH NAME
postnews \- Post a Usenet article to a server
.SH SYNOPSIS
.B postnews
.RI [ OPTIONS ]
.I SERVER
.SH DESCRIPTION
This manual page documents briefly the
.B postnews
command.
This manual page was written for the Debian distribution
because the original program does not have a manual page.
.PP
.B Postnews
is used to post Usenet articles (including headers) from stdin to a server.
The article must at least contain the headers
.IR From: ", " Newsgroups: ", " Subject: ,
a newline and a body.
.SH OPTIONS
These programs follow the usual GNU command line syntax, with long
options starting with two dashes ("-").
A summary of options is included below.
.TP
.BR \-h ", " \-\-help
Show summary of options.
.TP
.BR \-v ", " \-\-verbose
Be verbose.
.TP
.BR \-f ", " \-\-file= FILE
Read file instead of stdin.
.TP
.BR \-p ", " \-\-port
Port number.
.TP
.BR \-\-user
User name.
.TP
.BR \-\-pass
Password.
.TP
.BR \-r ", " \-\-readermode
Send MODE READER before authentication.
.SH AUTHOR
Postnews is written by
Michael Waschb\[:u]sch <waschbuesch@users.sourceforge.net>.
This manual page was written by Peter Krefting <peter@softwolves.pp.se>,
for the Debian project (but may be used by others).


ChangeLog
0.6.2
* Insert a docstring at line three of the postnews.py file.
* Add a space after the '!=' operator on line 31 of postnews.py.
* Remove unnecessary semicolon from line 35 of the postnews.py file.

0.6.1
* Expand the writeup in the README file regarding needed headers.
* Replace tabs with spaces and correct blank lines in postnews.py.
* Add 'license', 'platforms', & 'long_description' entries to the 'setup' stanza
  in the setup.py file.

0.6
* Add setup.py and MANIFEST.in
* Rename 'postnews' as 'postnews.py'.
* Update spelling of original authors last name.
* Correct copyright years to 2001-2002 for original author.
* Add Robert James Clay to the AUTHORS, README, and postnews.py files.

0.5.3
* Added the readermode option

0.5.2
* Changed the help message. (Thanks to Dan Jacobson)

0.5.1
* Minor bugfixes

0.5
* Made program quiet and added option '--verbose'
* Added some documentation

0.4
* Initial release


postnews.py
#!/usr/bin/env python
# coding: utf-8
"""Post a usenet article (including headers) from stdin onto SERVER."""

# postnews 0.6.2 - post a usenet article
#
# (C) 2001-2002 by Michael Waschbüsch <waschbuesch@users.sourceforge.net>
# (C) 2014-2016 Robert James Clay <jame@rocasa.us>
# http://sourceforge.net/projects/postnews/
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.

# Juhuu! Mein erstes Python-Programm :o)

import sys
import nntplib
import getopt


def main():
    #get arguments and options
    try:
        opts, args = getopt.getopt(sys.argv[1:], "hf:p:vr", ["help", "file=", "port=", "user=", "pass=", "verbose", "readermode"])
    except:
        usage()
        sys.exit(2)

    #parse arguments
    if len(args) != 1:
        usage()
        sys.exit(2)

    server = args[0]

    #parse options
    file = sys.stdin
    port = 119
    user = ""
    password = ""
    verbose = 0
    readermode = None

    for o, a in opts:
        if o in ("-h", "--help"):
            usage()
            sys.exit()
        if o in ("-f", "--file"):
            try:
                file = open(a)
            except IOError:
                sys.stderr.write("File not found: "+a+"\n")
                sys.exit(2)
        if o in ("-p", "--port"):
            try:
                port = int(a)
                if port < 0 or port > 65535:
                    raise ValueError
            except ValueError:
                sys.stderr.write("Invalid port number: "+a+"\n")
                sys.exit(2)
        if o == "--user":
            user = a
        if o == "--pass":
            password = a
        if o in ("-v", "--verbose"):
            verbose = 1
        if o in ("-r", "--readermode"):
            readermode = 1

    #post message
    if verbose:
        print "Connecting to Server..."
    try:
        s = nntplib.NNTP(server, port, user, password, readermode)
    except Exception, e:    # it can throw a class exception...
        sys.stderr.write("Can't connect to server: "+server+"\n")
        sys.stderr.write(str(e)+"\n")
        sys.exit(2)
    except:         # ... or a string exception
        sys.stderr.write("Can't connect to server: "+server+"\n")
        sys.stderr.write(sys.exc_info()[1]+"\n")
        sys.exit(2)
    if verbose:
        print "Posting article..."
    try:
        s.post(file)
    except Exception, e:    # it can throw a class exception...
        sys.stderr.write("Can't post the given input.\n")
        sys.stderr.write(str(e)+"\n")
        sys.exit(2)
    except:         # ... or a string exception
        sys.stderr.write("Can't post the given input.\n")
        sys.stderr.write(sys.exc_info()[1]+"\n")
        sys.exit(2)

    s.quit()


def usage():
    print "postnews 0.6.2"
    print " - (C) 2001-2002 by Michael Waschbüsch <MichaelWaschbuesch@web.de>"
    print " - (C) 2014-2016 by Robert James Clay <jame@rocasa.us>"
    print ""
    print "Usage: postnews [OPTIONS] SERVER"
    print "Post a usenet article (including headers) from stdin onto SERVER."
    print "Article must at least contain the headers 'From:', 'Newsgroups:' and 'Subject:',"
    print "a newline and a body."
    print ""
    print "Options: -h, --help          display this text"
    print "         -v, --verbose       be verbose"
    print "         -f, --file=FILE     read file instead of stdin"
    print "         -p, --port=PORT     port number"
    print "             --user=NAME     user name"
    print "             --pass=PASSWD   password"
    print "         -r, --readermode    send MODE READER before authentication"


if __name__ == "__main__":
    main()
